### DISCUSSION TITLE
IPFS support

### DISCUSSION THREAD

**Discussion by etozhecyber** [DblK/tinshop#25] [Ideas]
I think IPFS is perfect as a storage for big static files

**Comment by DblK** [comment]
Hi, I did not know about IPFS.
It could be a good addition but since I prefer to begin to add local storage, I will go for FTP first once NFS is finished before going for it.
But it will be a good addition. However if there is a lot of person wanting for this, I will change priorities.

Here is the library in Go (https://github.com/ipfs/kubo).

**Comment by DblK** [comment]
@etozhecyber I did some research to understand IPFS and I would like to understand how you think it should work with `tinshop`.

So, I have my backup on my laptop (or server) and I launch `tinshop`, what should happen next?

When the file should be added to IPFS? How can I know the ID of my file (once added I will have it) but discover won't be possible?
I bet I need to store the ID of all files?

Love to hear your though about it.
